tf5to6server/tf5to6server.go

```diff
@@ -0,0 +1,167 @@
+package tf5to6server
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-mux/internal/tfprotov5tov6"
+	"github.com/hashicorp/terraform-plugin-mux/internal/tfprotov6tov5"
+)
+
+// UpgradeServer wraps a protocol version 5 ProviderServer in a protocol
+// version 6 server. Protocol version 6 is fully forwards compatible with
```

Comment:
Certainly open to suggestions here, but the gist of this is that:

* Provider functionality in protocol version 5 maps 1:1 to available functionality in protocol version 6
* Protocol version 6 implements additional provider features on top of what was available in protocol version 5

So therefore it would be invalid to specify that protocol version 6 is "fully" backwards compatible, as it is not, however version 5 is forwards compatible.